python/ml_croissant/_src/operation_graph/base_operation.py

```diff
@@ -0,0 +1,29 @@
+"""Base operation module."""
+
+import dataclasses
+
+from ml_croissant._src.nodes import Node
+
+
+@dataclasses.dataclass(frozen=True, repr=False)
+class Operation:
```

Comment:
Suggestion: make this class abstract and make `__call__` an abstract method.
This would allow for the "implement abstract methods" function in IDEs when defining a new Operation